applications/nrf_desktop/src/events/config_event.h

```diff
@@ -52,130 +71,68 @@ extern "C" {
-	struct event_header header;
-
-	uint16_t recipient;
-	uint8_t id;
-	void *channel_id;
-};
-
-EVENT_TYPE_DECLARE(config_fetch_request_event);
-
-enum config_status {
-	CONFIG_STATUS_SUCCESS,
-	CONFIG_STATUS_PENDING,
-	CONFIG_STATUS_FETCH,
-	CONFIG_STATUS_TIMEOUT,
-	CONFIG_STATUS_REJECT,
-	CONFIG_STATUS_WRITE_ERROR,
-	CONFIG_STATUS_DISCONNECTED_ERROR,
-};
-
-/** @brief Configuration channel forward event.
- * Used to pass configuration from dongle to connected devices.
- */
-struct config_forward_event {
-	struct event_header header;
-
-	uint16_t recipient;
-	uint8_t id;
-	enum config_status status;
-
+	uint8_t event_id;
+	uint8_t status;
 	struct event_dyndata dyndata;
 };
 
-EVENT_TYPE_DYNDATA_DECLARE(config_forward_event);
-
-/** @brief Configuration channel forward get event.
- * Used to forward configuration channel get request to connected devices.
- */
-struct config_forward_get_event {
-	struct event_header header;
-
-	uint16_t recipient;
-	uint8_t id;
-	void *channel_id;
-	enum config_status status;
-};
-
-EVENT_TYPE_DECLARE(config_forward_get_event);
-
-/** @brief Configuration channel forwarded event.
- * Used to confirm that event has been successfully forwarded.
- */
-struct config_forwarded_event {
-	struct event_header header;
-
-	enum config_status status;
-};
+EVENT_TYPE_DYNDATA_DECLARE(config_event);
 
-EVENT_TYPE_DECLARE(config_forwarded_event);
+#ifdef CONFIG_USB_DEVICE_PID
+ #define DEVICE_PID CONFIG_USB_DEVICE_PID
+#elif defined(CONFIG_BT_GATT_DIS_PNP_PID)
+ #define DEVICE_PID CONFIG_BT_GATT_DIS_PNP_PID
+#else
+ #error Device PID is not defined
+#endif
 
 extern const uint8_t __start_config_channel_modules[];
 extern const uint8_t __stop_config_channel_modules[];
 
 #define GEN_CONFIG_EVENT_HANDLERS(mod_name, opt_descr, config_set_fn, config_fetch_fn, is_info)	\
 	BUILD_ASSERT(ARRAY_SIZE(opt_descr) > 0);						\
 	BUILD_ASSERT(ARRAY_SIZE(opt_descr) <= OPT_FIELD_MASK);					\
-	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {					\
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE) && is_config_event(eh)) {		\
 		static const uint8_t module_id_in_section					\
 			__attribute__((__section__("config_channel_modules"))) = 0;		\
 		uint8_t config_module_id =							\
 			&module_id_in_section - (uint8_t *)__start_config_channel_modules;	\
 		static uint8_t cur_opt_descr;							\
 												\
-		if (is_config_event(eh)) {							\
-			struct config_event *event = cast_config_event(eh);			\
+		struct config_event *event = cast_config_event(eh);				\
 												\
-			if (MOD_FIELD_GET(event->id) == config_module_id) {			\
-				__ASSERT_NO_MSG(config_set_fn != NULL); 			\
-				(*config_set_fn)(OPT_ID_GET(OPT_FIELD_GET(event->id)),		\
-						 event->dyndata.data,				\
-						 event->dyndata.size);				\
-			}									\
+		uint8_t rsp_data_buf[CONFIG_CHANNEL_FETCHED_DATA_MAX_SIZE];			\
+		size_t rsp_data_size = 0;							\
+		bool consume = false;								\
 												\
+		memset(rsp_data_buf, 0, sizeof(rsp_data_buf));					\
```

Comment:
That could be useful for debug purposes - for example if a module sets data size to bigger value than actually filled data or fills only part of the data, the rest of the response will have a fixed value of zeros instead of some "garbage" values.

Still, I could remove it or enable it only in debug configurations.